pkg/util/bs/bs.go

```diff
@@ -0,0 +1,19 @@
+package bs
+
+import (
+	"reflect"
+	"unsafe"
+)
+
+// String2Bytes converts string to byte slice without a memory allocation.
+func String2Bytes(s string) (b []byte) {
+	sh := *(*reflect.StringHeader)(unsafe.Pointer(&s))
```

Comment:
This is highly suspicious to me. If this was really safe to do, I would expect that go would do this internally when you do `[]byte()`. I could not find anything online suggesting this is ok to do, and there are no tests at all for this

Reply:
It is safe for the conversion from a string to a byte slice, the reason why `golang` didn't do it internally is that the byte slice converted from that way is immutable cuz `string` type is immutable, which doesn't make a common case for most of the byte-slices.

Reply:
Actually, `go` compiler did make that kind of optimizations when ranging over []byte(s), see:https://github.com/golang/go/wiki/CompilerOptimizations#range-over-bytes.